src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+
+    constructor(props) {
+        super(props);
+        this.state = {fields: FieldGroupUtils.getGroupFields(gkey)};
+    }
+
+    componentWillUnmount() {
+        this.iAmMounted = false;
+        if (this.unbinder) this.unbinder();
+    }
+
+    componentDidMount() {
+        this.iAmMounted = true;
+        this.unbinder = FieldGroupUtils.bindToStore(gkey, (fields) => {
+            if (fields !== this.state.fields && this.iAmMounted) {
+                this.setState({fields});
+            }
+        });
+    }
+
+    render() {
+        var {fields}= this.state;
+        return (
+            <div style={{padding: 10}}>
+                <FormPanel
+                    groupKey={gkey}
+                    onSubmit={(request) => onSearchSubmit(request)}
+                    onCancel={hideSearchPanel}>
+                    <CatalogSelectView fields={fields}/>
+                </FormPanel>
+            </div>
+        );
+    }
+}
+
+function onSearchSubmit(request) {
+    console.log('request ' + JSON.stringify(request));
+
+    if (request.Tabs === 'catalog') {
+        const wp = parseWorldPt(request[ServerParams.USER_TARGET_WORLD_PT]);
+        if (!wp && (request.spatial === SpatialMethod.Cone.value
+            || request.spatial === SpatialMethod.Box.value
+            || request.spatial === SpatialMethod.Elliptical.value)) {
+            showInfoPopup('Target is required');
+            return;
+        }
+        doCatalog(request);
+    }
+    else if (request.Tabs === 'loadcat') {
+        doLoadTable(request);
+    }
+    else if (request.Tabs === 'vosearch') {
+        doVoSearch(request);
+    }
+    else {
+        console.log('request no supported');
+    }
+}
+function hideSearchPanel() {
+    dispatchHideDropDown();
+}
+
+function doCatalog(request) {
+
+    const conesize = convertAngle('deg', 'arcsec', request.conesize);
+    var title = request.project + '-' + request.cattable + ' (' + request.spatial;
+    if (request.spatial === SpatialMethod.Box.value || request.spatial === SpatialMethod.Cone.value || request.spatial === SpatialMethod.Elliptical.value) {
+        title += ':' + conesize;
+    }
+    title += ')';
+    var tReq = TableRequest.newInstance({
+
+        id: 'GatorQuery',
+        title: title,
+        SearchMethod: request.spatial,
+        catalog: request.cattable,
+        RequestedDataSet: request.catalog,
+        use: 'catalog_overlay',
+        catalogProject: request.project
+
+    });
+
+    // change and merge others parameters in request if elliptical
+    // plus change spatial name to cone
+    // (Gator search method for elliptical is cone)
+    if (request.spatial === SpatialMethod.Elliptical.value) {
+
+        const pa = get(request, 'posangle', 0);
+        const ar = get(request, 'axialratio', 0.26);
+
+        // see PA and RATIO string values in edu.caltech.ipac.firefly.server.catquery.GatorQuery
+        merge(tReq, {'posang': pa, 'ratio': ar});
```

Comment:
I had to read about `merge`, I have never used it.  However, I don't think it is quite right here. You want to use `Object.assign`, you could use lodash `assign` but we try to choose the built in ones first.  lodash `merge` does inherited properties as well.  For a simple object literal it still will work but I don't think it is the right fit.